js/bitget3.js

```diff
@@ -0,0 +1,2899 @@
+            },
+        });
+    }
+
+    async fetchTime (params = {}) {
+        const response = await this.dataGetCommonTimestamp (params);
+        //
+        //     {
+        //         "ch":"",
+        //         "data":"1640187433920",
+        //         "errCode":"",
+        //         "errMsg":"",
+        //         "fail":"false",
+        //         "orderId":"",
+        //         "status":"ok",
+        //         "success":"true",
+        //         "ts":null
+        //     }
+        //
+        return this.safeInteger (response, 'data');
+    }
+
+    async fetchMarkets (params = {}) {
+        let types = this.safeValue (this.options, 'fetchMarkets');
+        if (!types.length) {
+            types = [
+                this.options['defaultType'],
+            ];
+        }
+        let result = [];
+        for (let i = 0; i < types.length; i++) {
+            const markets = await this.fetchMarketsByType (types[i], params);
+            result = this.arrayConcat (result, markets);
+        }
+        return result;
+    }
+
+    parseMarkets (markets) {
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            result.push (this.parseMarket (markets[i]));
+        }
+        return result;
+    }
+
+    parseMarket (market) {
+        //
+        // spot
+        //
+        //      {
+        //          "symbol": "BTC_USDT",
+        //          "status": "online",
+        //          "base_currency": "BTC",
+        //          "quote_currency": "USDT",
+        //          "tick_size": "2",
+        //          "size_increment": "4",
+        //          "base_asset_precision": "4"
+        //      }
+        //
+        // swap
+        //
+        //     {
+        //         "symbol":"btcusd",
+        //         "underlying_index":"BTC",
+        //         "quote_currency":"USD",
+        //         "coin":"BTC",
+        //         "contract_val":"1",
+        //         "listing":null,
+        //         "delivery":["07:00:00","15:00:00","23:00:00"],
+        //         "size_increment":"0",
+        //         "tick_size":"1",
+        //         "forwardContractFlag":false,
+        //         "priceEndStep":5,
+        //         "minLeverage":1,
+        //         "maxLeverage":125
+        //     }
+        //
+        const id = this.safeString (market, 'symbol');
+        let marketType = 'spot';
+        let spot = true;
+        let swap = false;
+        const baseId = this.safeString2 (market, 'base_currency', 'coin');
+        const quoteId = this.safeString (market, 'quote_currency');
+        const contractVal = this.safeNumber (market, 'contract_val');
+        if (contractVal !== undefined) {
+            marketType = 'swap';
+            spot = false;
+            swap = true;
+        }
+        const base = this.safeCurrencyCode (baseId);
+        const quote = this.safeCurrencyCode (quoteId);
+        let symbol = id.toUpperCase ();
```

Comment:
This is wrong. For swap markets, you are storing the exchange-specific id in the unified ccxt structure. 

Please check out here how the unified symbol should be constructed: https://docs.ccxt.com/en/latest/manual.html?highlight=swaps#perpetual-swaps-perpetual-futures

Reply:
Fixed in PR #11907